src/ReactShallowRenderer.js

```diff
@@ -500,7 +500,9 @@ See https://fb.me/react-invalid-hook-call for tips about how to debug and fix th
 
     this._rendering = true;
     this._element = element;
-    this._context = getMaskedContext(elementType.contextTypes, context);
+    this._context = element.contextType
```

Comment:
@unverbraucht shouldn't it be checking for `elementType.contextType` instead of `element.contextType`?

Reply:
I'm not sure, the approach in the PR worked for me. I'm afraid I can't test it anymore, we moved away from react-shallow-renderer completely on the project.

Reply:
Fair point if you've already moved on, which is what I'm already planning to do. However, I just confirmed that, while element.contextType is not defined, elementType.contextType is. Hopefully this little tidbit will help someone, even if this PR is never updated and merged.

Reply:
@ljharb I'm willing to update this PR since it's a blocker for https://github.com/enzymejs/enzyme/pull/2554. Operational wise, how should we do it? A branch on my fork? Another PR?

Reply:
@pablopalacios please post a link to a branch on your fork - NOT a separate PR - and i can pull its changes into this PR. Thanks!

Reply:
@ljharb https://github.com/pablopalacios/react-shallow-renderer/tree/add-context

Reply:
@unverbraucht @dcpesses after adding some tests, I tried both versions. `elementType` is the right one (and the one that makes sense since we extract the legacy context from it as well).